### ISSUE TITLE
New table of contents UI

### ISSUE THREAD

**Issue by Bosch-0** [BitcoinDesign/Guide#848]
We should adopt the below style for our table of contents (TOC) used in the guide. We should be using TOC a lot more often throughout the guide, it makes content much easier to navigate, especially on mobile. Imo the below style keeps things visually clean and is easy to navigate (on desktop it stays stickied to the right so its accessible no matter where you are on the page). Our current markdown styled TOC isn't visually as nice and the text is quite small (13px).

![image](https://user-images.githubusercontent.com/55287964/173485571-61ec1d0d-e324-4d10-8bc3-90ba4c97d734.png)

Screenshots from the Docusaurus site: https://docusaurus.io/docs

**Comment by Bosch-0** [comment]
> I could see that work well for super-wide screens

My 24" still show it without making things messy, this isn't a super wide resolution.

Yeah both fonts are a little small. 14px as a minimum I think is good for web.

**Comment by nityagupta6** [comment]
In my opinion this is a very helpful feature! I was thinking of implementing this, but after visiting the guild on my desktop I realized that we might not be able to add the toc at the right side of the page as in the example @Bosch-0 had put on because that won't suit the existing wireframe of the page. Instead, I have thought of 2 ways in which the toc could be integrated on the page, could someone pls suggest which one would look better? 

For the first idea, I was thinking of adding a very thin #F7931A colored column beside the left panel with a small right arrowhead at the center of that small column which when clicked slides from there and shows the toc on this page. Similar to how the menubar slides from the left when we click on a hamburger menu on a website just the difference here is that instead of the hamburger menu icon, there is a thin column (of #F7931A color) on the immediate right edge of the guild contents left panel with an arrowhead at its center. To see what I'm talking about you can have a look at the second last example on this website https://speckyboy.com/slide-out-sidebars/. For the responsive design, I think the way toc is implemented on the https://docusaurus.io/docs website is just perfect.

For the second idea I had thought that as soon as the user clicks on any page say for example 'wallet security' the left contents panel splits into 2 portions the upper portion for the toc for this page and the lower one for the toc of the guild. For the responsive design, I think the way toc is implemented on the https://docusaurus.io/docs website is just perfect.

Hope you liked the 2 ideas pls suggest which one suits better! And I would like to work on this issue to design the UI for toc, Please assign the issue to me.

**Comment by GBKS** [comment]
Thank you for the proposal. Overall, I think this solution is too complex. Imagine your design is implemented and your are a reader and land on this page. How would you know that the orange bar contains a hidden table of contents, since there is no indication? The orange bar draws a lot of attention, but since it only contains a small arrow, the user has to guess what it means. My first guess would have been that it expands the main navigation in the sidebar to fill the screen.

Another aspect is that this approach makes it less convenient to see the table of contents. Generally, we try to keep our pages pretty short. If a page gets too long, we try to split it up. So our tables of contents are usually short as well. Our current approach of simply having it on the page allows the user to scan it in a second without any further action. If we hide it in a second, expandable side-bar, they always have to go and find it. It's a lot of friction for seeing a list of 5 or so links.

I really think that a simple tweak here is enough that does not introduce lots of extra UI and the resulting complexity. That is just my feedback, if you feel strongly about your direction, feel free to continue exploring as you like.

**Comment by nityagupta6** [comment]
Yes, I can see the problems you have listed in this design, and so I have come up with a different solution!
![image](https://user-images.githubusercontent.com/97425446/235942781-00fadf71-87a7-47d8-824f-716f9cb47248.png)

Now as soon as the user clicks on an option in the sidebar navigation the page directly loads with the toc at first so that the user can go through the content on the page in the very beginning itself and then if he/she taps anywhere on the page or on the symbol for the toc the toc disappears. To have a look again at the toc the user can once again click on the toc icon to open up the toc. As in the below image.
![image](https://user-images.githubusercontent.com/97425446/235945356-e983f00b-6aef-4e90-8fab-9b0140b5c452.png)

@GBKS  What are your thoughts on this. I have tried to think of a better solution as per the problems you had pointed out on the previous design. Pls give your valuable suggestions🙂
